asb-ballerina/messageReceiver.bal

```diff
@@ -22,25 +22,36 @@ import ballerina/jballerina.java as java;
 public isolated client class MessageReceiver {
 
     final string connectionString;
-    final string entityPath;
+    final string queueName;
+    final string subscriptionName;
+    final string topicName;
     final string receiveMode;
     final handle receiverHandle;
 
     # Initializes the connector. During initialization you can pass the [Shared Access Signature (SAS) authentication credentials](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-sas)
-    # Create an [Azure account](https://docs.microsoft.com/en-us/learn/modules/create-an-azure-account/) and 
-    # obtain tokens following [this guide](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal#get-the-connection-string). 
+    # Create an [Azure account](https://docs.microsoft.com/en-us/learn/modules/create-an-azure-account/) and
+    # obtain tokens following [this guide](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal#get-the-connection-string).
     # Configure the connection string to have the [required permission](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-sas).
-    # 
-    # + connectionString - Connection String of Azure service bus
-    # + entityPath - Name or path of the entity (e.g : Queue name, Subscription path)
+    #
+    # + connectionString - Connection String of Azure service bus  
+    # + queueName - Queue name  
+    # + topicName - Topic name  
+    # + subscriptionName - Subscription name  
     # + receiveMode - Receive mode as PEEKLOCK or RECEIVEANDDELETE (default : PEEKLOCK)
-    public isolated function init(@display {label: "Connection String"} string connectionString, @display {label: "Entity Path"} string entityPath, 
-                                  @display {label: "Receive Mode"} string receiveMode = PEEKLOCK) returns error? {
+    # + return - Return Value Description
```

Comment:
Shall we move string receiveMode to an enum? Then the options are suggested to the user.